### PR TITLE
ADHOC: add read needle meta grpc

### PR DESCRIPTION
**Pull request by lyang24**
# What problem are we solving?
Add a grpc call/ resp to volume server to extract needle metadata (crc, ttl, last modified..) at specific offset with given size.

# How are we solving the problem?
added function to read needle at specific offset with provided size on volume servers.  ec shards not supported.

# How is the PR tested?
unit test added

# Checks
- [ ] I have added unit tests if possible.
- [ ] I will add related wiki document changes and link to this PR after merging.

### CODE REVIEW

File: weed/storage/erasure_coding/ec_volume.go [hunk 1]
```diff
@@ -210,10 +210,14 @@ func (ev *EcVolume) LocateEcShardNeedle(needleId types.NeedleId, version needle.
 		return types.Offset{}, 0, nil, fmt.Errorf("FindNeedleFromEcx: %v", err)
 	}
 
-	shard := ev.Shards[0]
+	intervals = ev.LocateEcShardNeedleInterval(version, offset.ToActualOffset(), types.Size(needle.GetActualSize(size, version)))
+	return
+}
 
+func (ev *EcVolume) LocateEcShardNeedleInterval(version needle.Version, offset int64, size types.Size) (intervals []Interval) {
+	shard := ev.Shards[0]
```
Comment (lyang24) [inline weed/storage/erasure_coding/ec_volume.go:218]: I am not sure if i can hardcode on to pick first shard here

File: weed/storage/store_ec.go [hunk 2]
```diff
@@ -162,6 +162,32 @@ func (s *Store) ReadEcShardNeedle(vid needle.VolumeId, n *needle.Needle, onReadS
 	return 0, fmt.Errorf("ec shard %d not found", vid)
 }
 
+func (s *Store) ReadEcShardNeedleAt(vid needle.VolumeId, n *needle.Needle, offset int64, size int32) (int, error) {
```
Comment (lyang24) [inline weed/storage/store_ec.go]: is it possible for me to pass a reader option that have flags that read meta data only

File: weed/storage/erasure_coding/ec_volume.go (hunk 1 above)
Comment (chrislusf) [reply weed/storage/erasure_coding/ec_volume.go:218]: do not need to support EC volumes.

**Comment by kmlebedev** [comment]
@lyang24 Thank you very much, I would also like to know more specifically, could this PR be useful?

**Comment by lyang24** [comment]
> 

Hi we will use this on volume.fsck command to fiter out files that modifed under the threshold  to avoid inconsistecies such as file chuck created on volume server but no information on filer yet
this is the related issue https://github.com/seaweedfs/seaweedfs/issues/2578

**Comment by kmlebedev** [comment]
> Hi we will use this on volume.fsck command to fiter out files that modifed under the threshold to avoid inconsistecies such as file chuck created on volume server but no information on filer yet this is the related issue #2578

Thanks, it seems this PR will also be useful for https://github.com/seaweedfs/seaweedfs/issues/3614

**Pull request merged**